v2/internal/util/metrictree/parser_test.go

```diff
@@ -283,14 +306,14 @@ func TestParseNamespaceElement_InvalidScenarios(t *testing.T) {
 		"  metr  ",
 		" metr",
 		"metr ",
-		"asd+",
+		"asd;",
 		"{asd[}",
 		"[group=]",
 		"[=id]",
 		"[=]",
-		"[gr+]",
+		"[gr^]",
```

Comment:
`^` has a special meaning for regexes, please use sth other than that

Reply:
this test checks invalid characters. ^ is used outside of regex, and it is forbidden

Reply:
ah, ok, I was misled by square brackets 👍